Loop/View Models/PresetsViewModel.swift

```diff
@@ -184,11 +184,53 @@ enum SelectablePreset: Hashable, Identifiable {
 public class PresetsViewModel {
 
     // MARK: Training
-    @ObservationIgnored @AppStorage("hasCompletedPresetsTraining") var hasCompletedTraining: Bool = false
```

Comment:
We can't just remove the @ObservationIgnored?

Reply:
Unfortunately property wrappers like `@AppStorage` do not play nicely with macros and it gives an error. The getter and setters just replicate the behavior the `@Observable` macro adds to each property not marked with `@ObservationIgnored`. 

https://dimillian.medium.com/the-sad-state-of-appstorage-and-the-observable-framework-9c924296a434

There's a library out there with a custom macro that can handle this automatically but I don't think it's necessary to add now. Hopefully Apple will make these systems play nicely together soon.

Reply:
One thing we can do if we find ourselves observing AppStorage frequently is centralizing them all into an `Observable` class. Since the observation will only trigger a view update if updating AppStorage from it's parent class. 

For example, if we're observing the same AppStorage property in 2 view models and then update the property in one of the viewModels, the other will not be able to observe that change.